packages/mui-base/src/Slider/Root/SliderRoot.types.ts

```diff
@@ -0,0 +1,294 @@
+
+export type Axis = 'horizontal' | 'vertical' | 'horizontal-reverse';
+
+export interface AxisProps<T extends Axis> {
+  offset: (
+    percent: number,
+  ) => T extends 'horizontal'
+    ? { left: string }
+    : T extends 'vertical'
+      ? { bottom: string }
+      : T extends 'horizontal-reverse'
+        ? { right: string }
+        : never;
+  leap: (
+    percent: number,
+  ) => T extends 'horizontal' | 'horizontal-reverse'
+    ? { width: string }
+    : T extends 'vertical'
+      ? { height: string }
+      : never;
+}
+
+export interface UseSliderReturnValue {
+  getRootProps: (
+    externalProps?: React.ComponentPropsWithRef<'span'>,
+  ) => React.ComponentPropsWithRef<'span'>;
+  /**
+   * The index of the active thumb.
+   */
+  active: number;
+  /**
+   * A function that compares a new value with the internal value of the slider.
+   * The internal value is potentially unsorted, e.g. to support frozen arrays: https://github.com/mui/material-ui/pull/28472
+   */
+  areValuesEqual: (newValue: number | ReadonlyArray<number>) => boolean;
+  'aria-labelledby'?: string;
+  /**
+   * The orientation of the slider.
+   */
+  axis: Axis;
+  changeValue: (
+    valueInput: number,
+    index: number,
+    event: React.KeyboardEvent | React.ChangeEvent,
+  ) => void;
+  compoundComponentContextValue: CompoundComponentContextValue<any, SliderThumbMetadata>;
+  dragging: boolean;
+  direction: SliderDirection;
+  disabled: boolean;
+  getFingerNewValue: (args: {
+    finger: { x: number; y: number };
+    move?: boolean;
+    offset?: number;
+    activeIndex?: number;
+  }) => { newValue: number | number[]; activeIndex: number; newPercentageValue: number } | null;
+  handleValueChange: (
+    value: number | number[],
+    activeThumb: number,
+    event: React.SyntheticEvent | Event,
+  ) => void;
+  /**
+   * The large step value of the slider when incrementing or decrementing while the shift key is held,
+   * or when using Page-Up or Page-Down keys. Snaps to multiples of this value.
+   * @default 10
+   */
+  largeStep: number;
+  /**
+   * The maximum allowed value of the slider.
+   */
+  max: number;
+  /**
+   * The minimum allowed value of the slider.
+   */
+  min: number;
+  /**
+   * The minimum steps between values in a range slider.
+   */
+  minStepsBetweenValues: number;
+  name?: string;
+  onValueCommitted?: (value: number | number[], event: React.SyntheticEvent | Event) => void;
+  /**
+   * The thumb index for the current value when in hover state.
+   */
+  open: number;
+  /**
+   * The component orientation.
+   * @default 'horizontal'
+   */
+  orientation: 'horizontal' | 'vertical';
+  registerSliderControl: (element: HTMLElement | null) => void;
+  /**
+   * The value(s) of the slider as percentages
+   */
+  percentageValues: readonly number[];
+  scale: (value: number) => number;
+  setActive: (activeIndex: number) => void;
+  setDragging: (isDragging: boolean) => void;
+  setOpen: (index: number) => void;
```

Comment:
What does "open" mean in context of a slider?

Reply:
This is set when the thumb is focused or hovered, I found it's only used by Material and Joy as an open state for the `ValueLabel` component

I'll remove it from here as this can be implemented externally like in this example: https://deploy-preview-373--base-ui.netlify.app/experiments/slider-tooltip